src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs

```diff
@@ -91,106 +91,101 @@ private Timestamp GetValidTimestamp(Signature signature, bool ignoreMultipleTime
-                            // if we are combining checks for more than one, then we have to use the whole list.
-                            IReadOnlyList<X509ChainStatus> chainStatus = null;
-                            if (SigningUtility.ChainStatusListIncludesStatus(chainStatusList, X509ChainStatusFlags.Revoked, out chainStatus))
+                            else if (!chainBuildingHasIssues)
                             {
-                                var status = chainStatus.First();
-                                issues.Add(SignatureLog.Issue(true, NuGetLogCode.NU3018, status.StatusInformation));
-                                return SignatureVerificationStatus.Invalid;
-                            }
-
-                            const X509ChainStatusFlags RevocationStatusFlags = X509ChainStatusFlags.RevocationStatusUnknown | X509ChainStatusFlags.OfflineRevocation;
-                            if (SigningUtility.TryGetStatusMessage(chainStatusList, RevocationStatusFlags, out messages))
-                            {
-                                if (failuresAreFatal)
-                                {
-                                    foreach (var message in messages)
-                                    {
-                                        issues.Add(SignatureLog.Issue(failuresAreFatal, NuGetLogCode.NU3018, message));
-                                    }
-                                }
-                                else if (!chainBuildingHasIssues)
-                                {
-                                    return SignatureVerificationStatus.Trusted;
-                                }
-                                chainBuildingHasIssues = true;
+                                return SignatureVerificationStatus.Trusted;
                             }
-
-                            // Debug log any errors
-                            issues.Add(SignatureLog.DebugLog(string.Format(CultureInfo.CurrentCulture, Strings.ErrorInvalidCertificateChain, string.Join(", ", chainStatusList.Select(x => x.ToString())))));
+                            chainBuildingHasIssues = true;
                         }
-                    }
-                    else
-                    {
-                        issues.Add(SignatureLog.Issue(failuresAreFatal, NuGetLogCode.NU3012, Strings.ErrorSignatureVerificationFailed));
+
+                        // Debug log any errors
+                        issues.Add(SignatureLog.DebugLog(string.Format(CultureInfo.CurrentCulture, Strings.ErrorInvalidCertificateChain, string.Join(", ", chainStatuses.Select(x => x.ToString())))));
                     }
                 }
                 else
                 {
-                    issues.Add(SignatureLog.Issue(failuresAreFatal, NuGetLogCode.NU3017, Strings.SignatureNotYetValid));
+                    issues.Add(SignatureLog.Issue(failuresAreFatal, NuGetLogCode.NU3000, Strings.ErrorSignatureVerificationFailed));
```

Comment:
I'd prefer not to use `NU3000` for this.  `NU3000` should be a fallback when we have no further context.  However, `issues` already contains some issues from validation.  Either eliminate this statement altogether or create a new `NuGetLogCode`.  I prefer the former option.